### PR TITLE
Fix and update GitHub release action

### PR DESCRIPTION
**Pull request by Philippus**
Applying this PR will hopefully fix the publishing workflow (https://github.com/sbt/sbt-header/issues/268). The sbt-ci-release plugin was missing. Also removed some unused plugins and updated some things in the workflows.

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -1,5 +1,3 @@
-addSbtPlugin("com.dwijnand"      % "sbt-dynver"   % "4.1.1")
-addSbtPlugin("com.dwijnand"      % "sbt-travisci" % "1.2.0")
-addSbtPlugin("de.heikoseeberger" % "sbt-header"   % "5.6.0")
-addSbtPlugin("org.foundweekends" % "sbt-bintray"  % "0.6.1")
-addSbtPlugin("org.scalameta"     % "sbt-scalafmt" % "2.4.6")
+addSbtPlugin("com.dwijnand"      % "sbt-ci-release" % "1.5.10")
+addSbtPlugin("de.heikoseeberger" % "sbt-header"     % "5.6.0")
```
Comment (hseeberger) [inline project/plugins.sbt:2]: Why "com.dwijnand"? The "official" group ID, according to https://github.com/sbt/sbt-ci-release, is "com.github.sbt".

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -8,12 +8,12 @@ on:
 
 jobs:
   build:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
```
Comment (hseeberger) [inline .github/workflows/test.yml:11]: No, please leave 20.04 for reproducible builds.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -8,12 +8,12 @@ on:
...
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-java@v2
         with:
-          distribution: adopt
-          java-version: 11
```
Comment (hseeberger) [inline .github/workflows/test.yml:17]: Leave version 11.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -6,12 +6,15 @@ on:
 
 jobs:
   publish:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
```
Comment (hseeberger) [inline .github/workflows/release.yml:9]: Leave 20.04.

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -6,12 +6,15 @@ on:
...
     steps:
       - uses: actions/checkout@v2
         with:
           fetch-depth: 0
-      - uses: olafurpg/setup-scala@v13
```
Comment (hseeberger) [inline .github/workflows/release.yml:14]: Revert this change, please, I do not want my projects to deviate too much.

**Review by hseeberger — CHANGES_REQUESTED** [review]

File: project/plugins.sbt (hunk 1 above)
Comment (Philippus) [reply project/plugins.sbt:2]: My bad, I didn't change the org.

**Comment by Philippus** [comment]
I've dropped the commit changing the workflow and fixed the org for sbt-ci-release.

**Comment by hseeberger** [comment]
Thanks! And I added the missing secretes (PGP key, etc.) to the repo. Looks like too many things were missing. ;-)
Thanks for your help!
